Data/Yaml/Internal.hs

```diff
@@ -172,6 +172,10 @@ textToValue _ _ t
   where x `isLike` ref = x `elem` [ref, T.toUpper ref, titleCased]
           where titleCased = toUpper (T.head ref) `T.cons` T.tail ref
 
+#if MIN_VERSION_aeson(0, 7, 0)
+textToScientific :: Text -> Either String Scientific
+textToScientific = Atto.parseOnly (Atto.scientific <* Atto.endOfInput)
```

Comment:
I thought aeson >= 0.7 would imply that we have an attoparsec with `Atto.scientific`, but this does not seem to be the case -- `aeson-0.7.0.0` has `attoparsec >= 0.11.1.0`, which does not have `Atto.scientific`.

With `attoparsec >= 0.11.1.0 && < 0.12` `aeson` used `Atto.rational` to parse `Scientific` numbers, which is not so good (see http://hackage.haskell.org/package/attoparsec-0.13.0.1/docs/Data-Attoparsec-Text.html#v:rational )

So I think we have to options:

* Add `attoparsec >= 0.12`
* More CPP to use `rational` if `attoparsec >= 0.11.1.0 && < 0.12`

I think the first option is the best, what do you think?